### ISSUE TITLE
Framework of extract Unity

### ISSUE THREAD

**Issue by pjcau** [jiulongw/swift-unity#106]
Hi Guys, 

so I do this tutorial and it's great job. Now i try to isolate this code in a module in my app, and I tryed to create a module(Framework in iOS). So in this point I blocked, because I have some error before.

So anybody can be or do this task before?

J

**Comment by ucerxxx** [comment]
Hi,

I believe separating Unity code in framework is the right way to go. I have experience in doing that. If community likes this idea I can help with it

**Comment by pjcau** [comment]
Thank for sharing my issue 👍. We can resolve it, if can be a dependency for the big library in cpp exported in library folder!!!

**Comment by armandsLa** [comment]
I agree on idea that exporting Unity is the best way to go. @ucerxxx do you have any information you would like to share on how to do it?

**Comment by ucerxxx** [comment]
Hi! I don't think it is a kind of some specific information). The approach I can suggest (and implement, but need to find some time to do that) is to move all unity generated code and initialization to the framework. The benefits - our project is much cleaner, we can put built framework to separate repository, save compilation time. The logic remains the same - we initialize Unity on app launch, then use it as usual. I am planning to find some time to start soon @armandsLa

**Comment by sayoojkrishnan** [comment]
Hey @ucerxxx , Any updates on this ?
